model/LGCN-TensorFlow/utility/load_data.py

```diff
@@ -270,24 +360,45 @@ def sample_neg_items_for_u(u, num):
             while True:
                 if len(neg_items) == num: break
                 neg_id = np.random.randint(low=0, high=self.n_items, size=1)[0]
-                if neg_id not in (self.test_set[u]+self.train_items[u]) and neg_id not in neg_items:
-                    neg_items.append(neg_id)
+                if self.alg_type in ['csgcn']:
+                    if neg_id not in [itemId for itemId,_ in (self.test_set[u]+self.train_items[u])] and neg_id not in neg_items:
+                        neg_items.append(neg_id)
+                else:
+                    if neg_id not in (self.test_set[u]+self.train_items[u]) and neg_id not in neg_items:
+                        neg_items.append(neg_id)
             return neg_items
     
         def sample_neg_items_for_u_from_pools(u, num):
             neg_items = list(set(self.neg_pools[u]) - set(self.train_items[u]))
             return rd.sample(neg_items, num)
 
-        pos_items, neg_items = [], []
         for u in users:
-            pos_items += sample_pos_items_for_u(u, 1)
-            neg_items += sample_neg_items_for_u(u, 1)
+            pos_items, neg_items = [], []
+            if self.alg_type in ['csgcn']:
+                pos_items_context, neg_items_context = [], []
+                pos_item_id, context = sample_pos_items_for_u(u, 1)[0]
+                neg_item_id = sample_neg_items_for_u(u, 1)[0]
+                pos_items.append(pos_item_id)
+                neg_items.append(neg_item_id)
+
+                pos_item_context, neg_item_context = [], []
```

Comment:
har vi vikke lige gjort alt det her på linje 314 og frem?

Reply:
Det her er for neg items, det andet er for pos